src/Schedule/TimePeriod.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Meringue\Schedule;
+
+use DateTimeImmutable as PHPDateTime;
+use Exception;
+use Meringue\FormattedDateTime\Date;
+use Meringue\FormattedDateTime\InCustomFormat;
+use Meringue\ISO8601DateTime;
+use Meringue\ISO8601DateTime\FromISO8601;
+use Meringue\Time;
+
+class TimePeriod
+{
+    private $from;
+    private $till;
+
+    public function __construct(Time $from, Time $till)
+    {
+        $this->from = $from;
+        $this->till = $till;
+
+        if ($this->from->greaterThan($this->till)) {
+            throw new Exception('Till time must be greater that from time. Next day must use multiply schedules.');
```

Comment:
"Till" must be greater than "from". Use a separate schedule object for the next day's schedule if you need to. See ByWeekDaysTest for details.